### DISCUSSION TITLE
Building tasmota with only needed sensors

### DISCUSSION THREAD

**Discussion by aik** [arendst/Tasmota#18281] [Q&A] [answered]
The goal is to skip the -minimal fw when upgrading. When I build tasmota-lite/minimal - these do not have any sensors even if I do
```
#define USE_DHT     
#define USE_I2C     
```
in `tasmota/user_config_override.h` (I expect AM2301/DHT11 in the config). On the other hand, when I build -sensors variant, they are all there and million more even though I add a lot of
```
#undef USE_SDM72
#undef USE_SDM120
#undef USE_SDM630
#undef USE_DDS2382
#undef USE_DDSU666
#undef USE_DHT
#undef USE_MAX31855
#undef USE_IR_REMOTE
#undef USE_IR_RECEIVE
#undef USE_LMT01
...
```
How do I build sensors-enabled tasmota with just 5 sensors supported? Thanks,

**Comment by barbudor** [comment, marked as answer]
As per the documentation you can only customize 'tasmota' and not any of the variants.
The defines for the variants are likely to override yous overrides. Only tasmota is meant to be customize for both adding and removing features
So you need to #undef what you don't want in your user_config_override.h and build 'tasmota'

The chance to achieve a firmware with sensors that is small enough to be flashable without minimal step are very low.

**Comment by aik** [comment]
btw with my current tasmota.bin.gz == 396,780 ("Program Size" is 552KB when it is flashed), I  can reflash it without the minimal step.

**Reply by barbudor** [reply]
Well done
